### ISSUE TITLE
Unsupported language for grammar 'YAML' with extension '.yml'.

### ISSUE THREAD

**Issue by s905060** [Glavin001/atom-beautify#312]
[Enter steps to reproduce below:]
1. ...
2. ...

**Atom Version**: 0.196.0
**System**: Mac OS X 10.10.3
**Thrown From**: [atom-beautify](https://github.com/Glavin001/atom-beautify) package, v0.26.4
### Stack Trace

Unsupported language for grammar 'YAML' with extension '.yml'.

```
At Unsupported language for grammar 'YAML' with extension '.yml'.

Error: Unsupported language for grammar 'YAML' with extension '.yml'.
  at /Users/jashl/.atom/packages/atom-beautify/src/beautifiers/index.coffee:351:28
  at tryCatcher (/Users/jashl/.atom/packages/atom-beautify/node_modules/bluebird/js/main/util.js:24:31)
  at Promise._resolveFromResolver (/Users/jashl/.atom/packages/atom-beautify/node_modules/bluebird/js/main/promise.js:427:31)
  at new Promise (/Users/jashl/.atom/packages/atom-beautify/node_modules/bluebird/js/main/promise.js:53:37)
  at Beautifiers.module.exports.Beautifiers.beautify (/Users/jashl/.atom/packages/atom-beautify/src/beautifiers/index.coffee:230:20)
  at Object.beautify (/Users/jashl/.atom/packages/atom-beautify/src/beautify.coffee:133:16)
  at TextBuffer.module.exports.Emitter.emit (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/emissary/lib/emitter.js:118:15)
  at TextBuffer.module.exports.TextBuffer.saveAs (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/text-buffer/lib/text-buffer.js:910:14)
  at TextBuffer.module.exports.TextBuffer.save (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/text-buffer/lib/text-buffer.js:899:19)
  at TextEditor.module.exports.TextEditor.save (/Applications/Atom.app/Contents/Resources/app.asar/src/text-editor.js:592:26)
  at Pane.module.exports.Pane.saveItem (/Applications/Atom.app/Contents/Resources/app.asar/src/pane.js:523:18)
  at Pane.module.exports.Pane.saveActiveItem (/Applications/Atom.app/Contents/Resources/app.asar/src/pane.js:506:19)
  at Workspace.module.exports.Workspace.saveActivePaneItem (/Applications/Atom.app/Contents/Resources/app.asar/src/workspace.js:585:35)
  at atom-workspace.atom.commands.add.core:save (/Applications/Atom.app/Contents/Resources/app.asar/src/workspace-element.js:310:30)
  at CommandRegistry.module.exports.CommandRegistry.handleCommandEvent (/Applications/Atom.app/Contents/Resources/app.asar/src/command-registry.js:238:29)
  at /Applications/Atom.app/Contents/Resources/app.asar/src/command-registry.js:3:61
  at KeymapManager.module.exports.KeymapManager.dispatchCommandEvent (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/atom-keymap/lib/keymap-manager.js:519:16)
  at KeymapManager.module.exports.KeymapManager.handleKeyboardEvent (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/atom-keymap/lib/keymap-manager.js:354:22)
  at HTMLDocument.module.exports.WindowEventHandler.onKeydown (/Applications/Atom.app/Contents/Resources/app.asar/src/window-event-handler.js:178:20)

```
### Commands

```
  3x -4:55.8.0 core:paste (atom-text-editor.editor)
     -3:57.3.0 core:save (atom-text-editor.editor)
  3x -3:24.3.0 core:paste (atom-text-editor.editor)
  2x -3:00.2.0 core:save (atom-text-editor.editor)
  2x -2:33.2.0 core:backspace (atom-text-editor.editor)
  4x -2:29.9.0 core:save (atom-text-editor.editor)
     -1:27.2.0 core:move-down (ol.tree-view.list-tree.has-collapsable-children.focusable-panel.full-menu)
     -1:04.1.0 core:move-left (atom-text-editor.editor.is-focused)
     -1:03.8.0 editor:newline (atom-text-editor.editor.is-focused)
     -0:36.3.0 core:copy (atom-text-editor.editor)
     -0:28.4.0 core:paste (atom-text-editor.editor.is-focused)
  2x -0:25.7.0 core:backspace (atom-text-editor.editor.is-focused)
     -0:23.2.0 core:save (atom-text-editor.editor.is-focused)
     -0:15 core:move-right (atom-text-editor.editor.is-focused)
  8x -0:14.6.0 core:backspace (atom-text-editor.editor.is-focused)
  2x -0:09.6.0 core:save (atom-text-editor.editor.is-focused)
```
### Config

``` json
{
  "core": {
    "themes": [
      "seti-ui",
      "monokai"
    ]
  },
  "atom-beautify": {
    "_analyticsUserId": "250400a8-e03f-4d36-ac7f-fb71bea392b6",
    "beautifyOnSave": true,
    "python_autopep8_path": "/usr/local/bin/autopep8",
    "python_indent_size": 4,
    "analytics": false,
    "beautifyEntireFileOnSave": false,
    "css_indent_size": 4
  }
}
```
### Installed Packages

``` coffee
# User
atom-beautify, v0.26.4
autoclose-html, v0.15.0
autocomplete-css, v0.6.0
autocomplete-html, v0.6.0
autocomplete-plus, v2.12.1
autocomplete-snippets, v1.3.0
compare-files, v0.4.1
highlight-selected, v0.9.2
indent-guide-improved, v1.1.1
linter, v0.12.1
linter-coffeelint, v0.2.2
linter-htmlhint, v0.0.13
linter-js-yaml, v1.0.0
linter-pylint, v0.2.1
linter-shellcheck, v0.0.7
minimap, v4.8.0
monokai, v0.14.0
seti-ui, v0.7.1

# Dev
No dev packages
```

**Comment by Glavin001** [comment]
YAML simply is not a supported language that can be beautified by Atom Beautify. I will be making this notification less aggressive with #317.

**Comment by Glavin001** [comment]
I'll look into YAML beautification support.

**Comment by opusmagnum** [comment]
Thanks in advance for YAML beautification support!

**Comment by Glavin001** [comment]
This may work: https://www.npmjs.com/package/align-yaml

YAML users please try and let me know if it works for you. If so, this beautifier could be added! :+1:

**Comment by gedw99** [comment]
i am using app engine and the yml files need to be like  "blahblah.yaml"

SO this does not work for me.
I just tried it.

**Comment by joshoohaah** [comment]
I agree the proposed alternative does not work.  An addition for beautify to add in YAML would be much appreciated!

**Comment by esetnik** [comment]
+1 let me know if I can help in any way. I'd love a yaml beautifier in atom.

**Comment by danbohea** [comment]
Also pretty keen to see YAML supported here. Thanks for the great package BTW :)

**Comment by JoshZA** [comment]
:+1:

**Comment by marcosscriven** [comment]
Just came here looking for YAML support too.

**Comment by cyrillk** [comment]
+1

**Comment by mjrk** [comment]
+1

**Comment by alrighttheresham** [comment]
+1

**Comment by fernandopro** [comment]
+1

**Comment by basnijholt** [comment]
+1

**Comment by aioue** [comment]
+1

**Comment by tomdavidson** [comment]
align-yaml works great for me so far. Hope to see this functionality in atom-beautify, in the mean time we can added it to atom save commands or what not.

**Comment by aioue** [comment]
@tomdavidson doesn't look like it sorts and/or preserves comment locations

**Comment by tomdavidson** [comment]
ahh, have not ran into that yet. Thanks for the heads up.

**Comment by polarathene** [comment]
Some form of alignment is better than none. Is everything else fine besides comments? There does not appear to be any atom package for aligning yaml easily. Would be nice to see the support added to atom-beautify.

**Comment by bushono** [comment]
+1

**Comment by Toub** [comment]
@Glavin001 any update about this?

**Comment by Glavin001** [comment]
I would be happy to review and merge a Pull Request adding YAML beautification support to Atom-Beautify.
Documentation for adding a language and beautifier to Atom-Beautify can be found at https://github.com/Glavin001/atom-beautify/blob/master/docs/add-languages-and-beautifiers.md 
It looks like `align-yaml` should work so I recommend using its API: https://github.com/jonschlinkert/align-yaml#api
Let me know if you have any questions. Thanks!

**Comment by Hronom** [comment]
Any update?

**Comment by JustinDevelops** [comment]
If I can make some time soon I am going to try and fork this project and add in the align-yaml from @Glavin001 recommendation. It might not be until after the holidays but I would really like to do this. 

I use YAML daily in Atom for AWS stuff, being able to beautify it so I don't get dumb formatting errors in CloudFormation.

**Comment by robert-john-small** [comment]
+1

**Comment by Isarstyle** [comment]
+1

**Comment by Trandel** [comment]
+1

**Comment by Glavin001** [comment]
This would be great and I am definitely interested in reviewing and merging a Pull Request implementing YAML support. Please let me know if you have any questions about adding languages and beautifiers to Atom-Beautify!

**Comment by jr00n** [comment]
+1

**Comment by nchinan** [comment]
+1

**Comment by Shinigami92** [comment]
+1

**Comment by cesasol** [comment]
+1

**Comment by basnijholt** [comment]
Please stop with the +1. Give the original post a thumps up instead.

**Comment by michaelgilhooly** [comment]
Has there been any progress on this feature? Really looking forward to it.

**Comment by polarathene** [comment]
I just tried align-yaml and it works fine with the example. I don't have any more extensive yaml examples to test with. I could probably follow the atom-beautify docs to add support for it but with the efforts to transition towards [Unibeautify](https://github.com/Glavin001/atom-beautify/issues/1174), I'm not sure if there is any point? For now it might be better to use one of the many online ones that you can copy/paste to.

I don't presently have the motivation to work on this as I haven't worked with yaml in a while, nor does it seem worth the effort. If someone is interested in working on it or would contribute support for Unibeautify instead, perhaps you can help by providing a decent yaml sample for tests(required part of the process for submission).

**Comment by Glavin001** [comment]
@polarathene since you have confirmed [align-yaml](https://github.com/jonschlinkert/align-yaml#api) works well enough then we can add it to Atom-Beautify. I am still accepting Pull Requests for Atom-Beautify pre-Unibeautify and I will likely personally migrate them all over to Unibeautify when I am ready. Thank you all for your patience!

**Comment by polarathene** [comment]
@Glavin001 I didn't have any other yaml source to try with beyond the example given on the repo that handles alignment. That worked fine however :)

**Comment by Glavin001** [comment]
Awesome! Pull Requests welcome: https://github.com/Glavin001/atom-beautify/blob/master/docs/add-languages-and-beautifiers.md
Please let me know if you have any questions.

**Comment by Glavin001** [comment]
Since I received no Pull Request, I implemented it on my own: https://github.com/Glavin001/atom-beautify/pull/1600

Will be in the next release.

**Comment by Glavin001** [comment]
Published to v0.29.22
